lib/puppet/util/windows/monkey_patches/dir.rb

```diff
@@ -0,0 +1,418 @@
+require 'puppet/util/windows'
+
```

Comment:
I know this is moved from the win32-dir gem but i think we can improve a bit the readability by moving class methods here in a `class << self` and avoid using alternatively def `self.`. And have the `private` inside `class << self` where we define all the private methods. No functionality changes, just a better organisation of the code.